### ISSUE TITLE
"WARNING: An illegal reflective access operation has occurred" during gem build (from jopenssl)

### ISSUE THREAD

**Issue by PragTob** [jruby/jruby#6356]
Hey there, this is over on https://github.com/colszowka/simplecov / https://github.com/colszowka/simplecov/issues/908

also naturally as always, thanks a bunch my friends for your great work! :green_heart: 

I should be on the newest version of everything. I did look at #5864 and #6049 so decided to open a new one.

```
tobi@speedy:~/github/simplecov(master)$ uname -a
Linux speedy 5.4.0-42-generic #46~18.04.1-Ubuntu SMP Fri Jul 10 07:21:24 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
tobi@speedy:~/github/simplecov(master)$ jruby -v
jruby 9.2.11.1 (2.5.7) 2020-03-25 b1f55b1a40 OpenJDK 64-Bit Server VM 14.0.2+12 on 14.0.2+12 +jit [linux-x86_64]
tobi@speedy:~/github/simplecov(master)$ java -version
openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment AdoptOpenJDK (build 14.0.2+12)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.2+12, mixed mode, sharing)
tobi@speedy:~/github/simplecov(master)$ bundle exec jruby -J--illegal-access=debug -S gem build simplecov.gemspec 
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method sun.nio.ch.NativeThread.signal(long)
WARNING: Please consider reporting this to the maintainers of com.headius.backport9.modules.Modules
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/home/tobi/.asdf/installs/ruby/jruby-9.2.11.1/lib/ruby/stdlib/jopenssl.jar) to field java.security.MessageDigest.provider
	at org.jruby.ext.openssl.SecurityHelper.setField(SecurityHelper.java:854)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:386)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:359)
	at org.jruby.ext.openssl.Digest.getDigest(Digest.java:123)
	at org.jruby.ext.openssl.Digest.initializeImpl(Digest.java:167)
	at org.jruby.ext.openssl.Digest$Named.initialize(Digest.java:346)
	at org.jruby.ext.openssl.Digest$Named$INVOKER$i$0$1$initialize.call(Digest$Named$INVOKER$i$0$1$initialize.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:811)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:365)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:153)
	at org.jruby.dist/org.jruby.RubyClass.newInstance(RubyClass.java:911)
	at org.jruby.dist/org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroOrNBlock.call(JavaMethod.java:327)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:354)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:143)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80)
	at org.jruby.dist/org.jruby.ir.instructions.DefineModuleInstr.INTERPRET_MODULE(DefineModuleInstr.java:81)
	at org.jruby.dist/org.jruby.ir.instructions.DefineModuleInstr.interpret(DefineModuleInstr.java:69)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.processOtherOp(StartupInterpreterEngine.java:178)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:104)
	at org.jruby.dist/org.jruby.ir.interpreter.Interpreter.INTERPRET_ROOT(Interpreter.java:96)
	at org.jruby.dist/org.jruby.ir.interpreter.Interpreter.execute(Interpreter.java:81)
	at org.jruby.dist/org.jruby.ir.interpreter.Interpreter.execute(Interpreter.java:30)
	at org.jruby.dist/org.jruby.ir.IRTranslator.execute(IRTranslator.java:42)
	at org.jruby.dist/org.jruby.Ruby.runInterpreter(Ruby.java:1218)
	at org.jruby.dist/org.jruby.Ruby.loadFile(Ruby.java:2785)
	at org.jruby.dist/org.jruby.runtime.load.LibrarySearcher$ResourceLibrary.load(LibrarySearcher.java:234)
	at org.jruby.dist/org.jruby.runtime.load.LibrarySearcher$FoundLibrary.load(LibrarySearcher.java:34)
	at org.jruby.dist/org.jruby.runtime.load.LoadService.tryLoadingLibraryOrScript(LoadService.java:887)
	at org.jruby.dist/org.jruby.runtime.load.LoadService.smartLoadInternal(LoadService.java:535)
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/home/tobi/.asdf/installs/ruby/jruby-9.2.11.1/lib/ruby/stdlib/jopenssl.jar) to field java.security.MessageDigest.provider
	at org.jruby.ext.openssl.SecurityHelper.setField(SecurityHelper.java:854)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:386)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:359)
	at org.jruby.ext.openssl.Digest.getDigest(Digest.java:123)
	at org.jruby.ext.openssl.Digest.initializeImpl(Digest.java:167)
	at org.jruby.ext.openssl.Digest$Named.initialize(Digest.java:346)
	at org.jruby.ext.openssl.Digest$Named$INVOKER$i$0$1$initialize.call(Digest$Named$INVOKER$i$0$1$initialize.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:811)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:365)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:153)
	at org.jruby.dist/org.jruby.RubyClass.newInstance(RubyClass.java:911)
	at org.jruby.dist/org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroOrNBlock.call(JavaMethod.java:327)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:354)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:143)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.dist/org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116)
	at org.jruby.dist/org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136)
	at org.jruby.dist/org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:166)
	at org.jruby.dist/org.jruby.runtime.BlockBody.yield(BlockBody.java:108)
	at org.jruby.dist/org.jruby.runtime.Block.yield(Block.java:184)
	at org.jruby.dist/org.jruby.RubyArray.collect(RubyArray.java:2563)
	at org.jruby.dist/org.jruby.RubyArray.map19(RubyArray.java:2577)
	at org.jruby.dist/org.jruby.RubyArray$INVOKER$i$0$0$map19.call(RubyArray$INVOKER$i$0$0$map19.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:555)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:332)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:86)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:93)
	at org.jruby.dist/org.jruby.ir.instructions.CallBase.interpret(CallBase.java:549)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/home/tobi/.asdf/installs/ruby/jruby-9.2.11.1/lib/ruby/stdlib/jopenssl.jar) to field java.security.MessageDigest.provider
	at org.jruby.ext.openssl.SecurityHelper.setField(SecurityHelper.java:854)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:386)
	at org.jruby.ext.openssl.SecurityHelper.getMessageDigest(SecurityHelper.java:359)
	at org.jruby.ext.openssl.Digest.getDigest(Digest.java:123)
	at org.jruby.ext.openssl.Digest.initializeImpl(Digest.java:167)
	at org.jruby.ext.openssl.Digest$Named.initialize(Digest.java:346)
	at org.jruby.ext.openssl.Digest$Named$INVOKER$i$0$1$initialize.call(Digest$Named$INVOKER$i$0$1$initialize.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodN.call(JavaMethod.java:811)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:151)
	at org.jruby.dist/org.jruby.RubyClass.newInstance(RubyClass.java:911)
	at org.jruby.dist/org.jruby.RubyClass$INVOKER$i$newInstance.call(RubyClass$INVOKER$i$newInstance.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroOrNBlock.call(JavaMethod.java:327)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:354)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:143)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:345)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.dist/org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:116)
	at org.jruby.dist/org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:136)
	at org.jruby.dist/org.jruby.runtime.IRBlockBody.doYield(IRBlockBody.java:166)
	at org.jruby.dist/org.jruby.runtime.BlockBody.yield(BlockBody.java:108)
	at org.jruby.dist/org.jruby.runtime.Block.yield(Block.java:184)
	at org.jruby.dist/org.jruby.RubyArray.collect(RubyArray.java:2563)
	at org.jruby.dist/org.jruby.RubyArray.map19(RubyArray.java:2577)
	at org.jruby.dist/org.jruby.RubyArray$INVOKER$i$0$0$map19.call(RubyArray$INVOKER$i$0$0$map19.gen)
	at org.jruby.dist/org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:555)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:84)
	at org.jruby.dist/org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:93)
	at org.jruby.dist/org.jruby.ir.instructions.CallBase.interpret(CallBase.java:549)
	at org.jruby.dist/org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:361)
	at org.jruby.dist/org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.dist/org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:86)
	at org.jruby.dist/org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:73)
  Successfully built RubyGem
  Name: simplecov
  Version: 0.18.5
  File: simplecov-0.18.5.gem
```

Thanks a lot!

**Comment by mistergibson** [comment]
I'm having very similar troubles on my end using OpenJDK 14 and the latest JRuby:

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/opt/jruby/lib/ruby/stdlib/jopenssl.jar) to field java.security.MessageDigest.provider
WARNING: Please consider reporting this to the maintainers of org.jruby.ext.openssl.SecurityHelper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

**Comment by headius** [comment]
@mistergibson The warning is just a warning, so unless the output itself is causing problems you should not have to worry about it.

This will is fixed for JRuby 9.2.14 and 9.3 with the updated jruby-openssl from #6423.
